### ISSUE TITLE
lr_scheduler step() called after every epoch instead of after every batch

### ISSUE THREAD

**Issue by ErikEngerd** [PyTorchLightning/pytorch-lightning#5631]
## 🐛 Bug

When using a learning rate scheduler, the scheduler's step() method is called once at the start of every epoch() and not, as mentioned in the docs, at the start of every batch. This leads to a constant learning rate per epoch when using for instance the CyclicLR scheduler. 

## Please reproduce using the BoringModel

[BoringModel](https://colab.research.google.com/drive/1yHaDmHSFFWjv2R90U7Tvp9Ek1KgYA2-c?usp=sharing)

### To Reproduce

Run the notebook. See the comments before the test is being done and examine the output. 

### Expected behavior

- lr_scheduler step() is called every batch and not every epoch
- learning rate changes every batch. 

### Environment

* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     True
	- pyTorch_version:   1.7.0+cu101
	- pytorch-lightning: 1.1.5
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.6.9
	- version:           #1 SMP Thu Jul 23 08:00:38 PDT 2020

### Additional context

-

**Comment by awaelchli** [comment]
https://pytorch-lightning.readthedocs.io/en/stable/lightning_module.html#configure-optimizers

I fixed it for you:
https://colab.research.google.com/drive/1yHaDmHSFFWjv2R90U7Tvp9Ek1KgYA2-c?usp=sharing

**Comment by ErikEngerd** [comment]
Thanks for your quick response. So it is not a code issue.

I guess the confusion is between the docs of pytorch that mention for CyclicLR that  step() must be called every batch, the generatl pytorch docs that talk about iterations (not batches/epochs), and the pytorch lightning docs that have generalized a bit and allow update every n batches or epochs.

**Comment by awaelchli** [comment]
Yes you can also call the scheduler manually yourself and not rely on Lightning. 
However keep in mind that you may have to write much more boilerplate code to cover all cases. 

One just has to know the terminology, some lr_schedulers define an epoch as the number of steps it takes before an update is made, this is then different from a training epoch. 
It's just a matter of defintion.

**Comment by ErikEngerd** [comment]
I want to avoid writing boilerplate. It is now working as expected. I am coming from tensorflow/keras, where I ran into a lot of issues (memory leaks, memory usage in general, tensorflow addons not working, complicated dataset code, etc.). I could net get cyclic and differential learning rates to work with tensorflow/keras. So far, I am liking the pytorch lightning APIs very much. The model of pytorch/pyrtoch lightning is much more clean and provides lots of ways to debug. Debugging is easier since the existing code is easier to understand and extend. It looks like pytorch lightning is at the right abstraction level.
